vm/devices/storage/disk_get_vmgs/src/lib.rs

```diff
@@ -47,76 +62,95 @@ impl VmgsGet {
     /// Failing to do so may result in data corruption/loss, though, notably: it
     /// will _not_ result in any memory-unsafety (hence why the function isn't
     /// marked `unsafe`).
-    pub fn new_with_meta(get: GuestEmulationTransportClient, meta: BlockStorageMetadata) -> Self {
-        VmgsGet { get, meta }
+    pub fn new_with_meta(
```

Comment:
if we have enough changes for another iteration, I'd want to change this method name to `dangerous_new_with_meta`, and tweak the `# Safety` block to `# Danger` instead.

i.e: make it a bit more obvious that even though there aren't memory safety issues when using the method, callers must still take special care to ensure metadata matches what is expected